backend/tests/data_layer/data_saver/test_jsonl_saver.py

```
@@ -40,145 +39,665 @@ def jsonl_config() -> DictConfig:
+    # Mock open to raise PermissionError
+    mocker.patch("builtins.open", side_effect=PermissionError("Permission denied"))
+
+    # Use temp_dir to avoid permission issues with directory creation
+    restricted_file = Path(temp_dir) / "restricted.jsonl"
+    saver = JSONLDataSaver(mock_consumer, restricted_file)
+    saver.retrieve_and_save()
+
+    error_calls = mock_logger.error.call_args_list
+    assert (
+        "call('Error while saving data to jsonl: %s', PermissionError('Permission denied'))"
+        == str(error_calls[0])
```

Ruff (0.11.9)

Yoda condition detected

(SIM300)